invenio_vocabularies/contrib/names/permissions.py

```diff
@@ -10,11 +10,23 @@
 
 from invenio_records_permissions.generators import AuthenticatedUser, SystemProcess
 
-from ...services.permissions import PermissionPolicy
+from invenio_vocabularies.services.generators import IfTags
+from invenio_vocabularies.services.permissions import PermissionPolicy
 
 
 class NamesPermissionPolicy(PermissionPolicy):
-    """Permission policy."""
+    """Names permission policy.
 
-    can_search = [SystemProcess(), AuthenticatedUser()]
-    can_read = [SystemProcess(), AuthenticatedUser()]
+    Names endpoints are protected, only authenticated users can access them.
+    """
+
+    can_search = [
+        SystemProcess(),
+        IfTags(exclude=["unlisted"], only_authenticated=True),
+    ]
```

Comment:
```suggestion
    can_search = [
        SystemProcess(),
        IfTags(exclude=["unlisted"]),
        AuthenticatedUser(),
    ]
```

minor: wouldn't that be a bit possible in terms of reusing/composing with existing generators? The `only_authenticated` parameter feels a bit ad-hoc

Reply:
Right! I agree but unfortunately it doesn't work like that, at the moment if we change it to your suggestion (which was my first thought/try as well) it will simply grant access if any of the needs is matched, meaning that ti will match for unlisted OR authenticated, while what we want to achieve is to grant access to unlisted AND authenticated for Names vocab while for the rest we want to grant access to unlisted AND any user.

That's why I added that param. The only easy alternative I can think of is creating 2 generators that would be `AnyUserIfTags` and `AuthenticatedUserIfTags` maybe this would be more readable ... Unless there are better sugesstions, they are more than welcomed!

Reply:
I also agree that is not idea to "hardcode" the authenticated part in a `IfTags`.
I would rather specialize the unlisted instead, which is a bit different, as only admins should be able to see it. Let's discuss IRL.

Reply:
Discussed IRL and agreed to use a generator like this, change implemented
```python
IfTags(["unlisted"], then_=[SystemProcess()], else_=[AnyUser()])
```